### ISSUE TITLE
Generated .app doesn't show up, but unix executable works fine

### ISSUE THREAD

**Issue by adamsss92** [pyinstaller/pyinstaller#4004]
Hello,

I'm trying to compile my python script (with tkinter GUI) to .app using pyinstaller with following arguments: 
- -D, --onedir
- -F, --onefile
- w, --windowed, --noconsole
and with icon.
I've tested many combinations of pyinstaller arguments, but it doesn't help.

There are unix executable file and .app file in "dist" folder. When i try to run .app file by doubleclicking or with terminal "open /~~~.app" application will not show up. I see that .app is launching on the Dock, but it immediately closed after few seconds. 

When i run the unix executable file everything works fine and GUI is showed up. Of course with terminal in the background.

Do you have an idea how to fix it?

My specification:
Python 3.5.2
macOS Sierra  10.12

Best regards,
Adam

**Comment by htgoebel** [comment]
Please provide more details, see [How to report Bugs](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs). Esp. please provide a *minimal* example to reproduce the error.

**Comment by xuxin19941214** [comment]
I also have this problem: the terminal can be opened, the program will be displayed, click on the program will flash directly, please ask the landlord to solve it?

**Comment by xuxin19941214** [comment]
> I think `-D` is interfering with `-F`.
> 
> Try:
> 
> ```
> pyinstaller -y --clean -F -w ./script.py
> ```
> Or if you have an icon:
> 
> ```
> pyinstaller -y --clean -F -w -i ./icon.icns ./script.py
> ```
> Works well for me.
> Hope that helps. Let me know how it goes.

Use your method or the same result, click on the program can not open, but the terminal open, program display, but can not let the terminal display, trouble

**Comment by htgoebel** [comment]
>There are unix executable file and .app file in "dist" folder. When i try to run .app file by doubleclicking or with terminal "open /~~~.app" application will not show up. I see that .app is launching on the Dock, but it immediately closed after few seconds.
>
>When i run the unix executable file everything works fine and GUI is showed up. Of course with terminal in the background.

This sound like you building with two options changed: `-D --console` vs. `-F --noconsole`

> I think -D is interfering with -F.

Of course, they are mutual exclusive.

> Works well for me.

Fine then. What is the problem now?

**Comment by L0y3r** [comment]
I'm having this issue too, have I to report it in another issue?

**Comment by Legorooj** [comment]
@L0y3r could yoy provide more details?

**Comment by L0y3r** [comment]
> @L0y3r could yoy provide more details?

@Legorooj  I already resolved the issue, my project depends on selenium and I bundle it with `--onefile` and `--windowed` flags, and when I tried to run from the .app folder it doesn't start the program, but the terminal executable can start as a normal app

**I fixed it, specifying the path of `service_log_path` pointing to `f'{sys._MEIPASS}/geckodriver.log'`**

**Comment by Legorooj** [comment]
Closing then.

**Comment by jaywalkerpaz** [comment]
I was having the same issue. I run: pyinstaller -w --windowed --noconsole --onefile main.spec and didn't get an app file. I ran pyinstaller on my py file directly and then got an app file, but saw my spec file was overwritten. When I rebuilt my spec file I'm now getting the .app file. I don't know why this happened, but maybe a typo in my spec or something hosing it up? Anyway just in case others get this issue
